oc_art_to_ftp/app/art_to_ftp.py

```diff
@@ -0,0 +1,388 @@
+                self._ftp_dir_create(current_path)
+
+        logging.debug('Paths created successfully')
+            
+    def _ftp_path_exists(self, path):
+        """
+        """
+        logging.debug('Reached _ftp_path_exists')
+        logging.debug('path: [%s]' % path)
+        ftp = self._ftp_connect()
+        try:
+            i = ftp.getinfo(path)
+        except fs.errors.ResourceNotFound as e:
+            logging.debug('Path does not exist')
+            return False
+        return i.is_dir
+
+    def _ftp_path_from_gav(self, gav):
+        """
+        """
+        logging.debug('Reached _ftp_path_from_gav')
+        logging.debug('gav: [%s}' % gav)
+        logging.debug('stripping gav')
+        gav = gav.strip()
+        client_code = self._client_code_from_gav(gav)
+        filename = NexusAPI.gav_to_filename(gav)
+        logging.debug('client_code: [%s]' % client_code)
+        logging.debug('filename: [%s]' % filename)
+        if client_code:
+            path = posixpath.join(posixpath.sep, client_code, 'TO_BNK', filename)
+            logging.debug('path: [%s]' % path)
+            return path
+        logging.error('Failed to get client_code')
+        return None
+ 
+    def _gav_to_upload(self, gav):
+        """
+        """
+        logging.debug('Reached _gav_to_upload')
+        logging.debug('gav: [%s]' % gav)
+        if ':customization-' in gav:
+            logging.debug('this gav is to be uploaded')
+            return True
+        return False
+
+    def _init_svn(self):
+        """
+        """
+        logging.debug('Reached _init_svn')
+        svn_url = os.getenv('SVN_URL')
+        svn_user = os.getenv('SVN_USER')
+        svn_pass = os.getenv('SVN_PASSWORD')
+
+        svn_client_fs = get_svn_fs_client(svn_url, svn_user, svn_pass)
+
+        return svn_client_fs
+
+    def _ls_artifactory(self, location, mask=None):
+        logging.debug('Reached _ls_artifactory')
+        logging.debug('location: [%s]' % location)
+        logging.debug('mask: [%s]' % mask)
+        na = self.nexus_api
+        if mask is None: mask = '.*'
+        files = na.ls(mask, repo = location)
+        logging.debug('na.ls returned: %s' % files)
+        return files
+
+    def _media_is_supported(self, media):
+        """
+        """
+        logging.debug('Reached _media_is_supported')
+        logging.debug('media: [%s]' % media)
+
+        if media.lower() not in self.supported_media:
+            logging.debug('Media is not supported')
+            logging.debug('Currently supported are: %s' % self.supported_media)
+            return False
+        
+        logging.debug('Media is supported')
+        return True
+
+    def _process_artifact(self, gav, target_path):
+        logging.debug('Reached _process_artifact')
+        data = tempfile.TemporaryFile()
+        na = self.nexus_api
+        logging.debug('Tempfile object is [%s]' % data)
+        na.cat(gav, binary=True, write_to=data)
```

Comment:
Please add `stream=True`.
Downloading large files without it may cause `out of memory` exception.